src/main/java/bio/terra/common/kubernetes/KubePodListener.java

```diff
@@ -80,6 +80,7 @@
   }
 
   @Override
+  @SuppressWarnings({"CyclomaticComplexity", "ExecutableStatementCount", "NPathComplexity"})
```

Comment:
I agree this method does too much, but I also don't really want to have to deal with stylistic warnings. Why not exclude these as well? I think this is a problem for code review, not a problem for automated analysis.

Reply:
I'd file a ticket to look at the method if it's tripping these complexity thresholds, or at least comment why it must be this complex/all one method.

Reply:
In light of the principles we've drawn up, I'm opting to exclude these rules.